src/services/address.service.ts

```
@@ -35,36 +36,211 @@ class AddressService {
+                    loggerService.error({ path: "/address/distance", message: errMessage }).flush();
+                    return { error: errMessage };
+                }
+                
+                let averageCoordinates1 = await this.calculateAverageCoordinates(location1);
+                let averageCoordinates2 = await this.calculateAverageCoordinates(location2);
+                
+
+                if (!averageCoordinates1 || !averageCoordinates2) {
+                    const errMessage = "Invalid locations. Please ensure both locations are valid."
+                    loggerService.error({ path: "/address/distance", message: errMessage }).flush();
+                    return { error: errMessage };
+                }
+
+                if (addressRequest.body.unit === "KM") {
+                    
+                    const distance = await this.getKilometerDistance(averageCoordinates1.latitude, averageCoordinates1.longitude, averageCoordinates2.latitude, averageCoordinates2.longitude);
+                    return { distance: distance };
+
+                } else if (addressRequest.body.unit === "Mi") {
+                    const distance = await this.getMilesDistance(averageCoordinates1.latitude, averageCoordinates1.longitude, averageCoordinates2.latitude, averageCoordinates2.longitude);
+                    return { distance: distance };
+
+                } else {
+                    const errMessage = "Invalid unit. Please use 'KM' or 'Mi'."
+                    loggerService.error({ path: "/address/distance", message: errMessage }).flush();
+                    return { error: errMessage };
+                }
+
+
+            } catch (err) {
+                loggerService.error({ path: "/address/distance", message: `${(err as Error).message}` }).flush();
+                return { error: (err as Error).message };
+            }
+        }
+
+        try {
+
+
```

I've seen empty catch blocks but not sure I've seen an empty try block before

---

It's a load bearing try block! (Can be removed, missed it on my own pass through when cleaning up)